src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,184 @@
+        ui.out("Bye. Hope to see you again soon!");
+        ui.close();
+        return SHUTDOWN;
+    }
+
+    private boolean list() {
+        ui.out(tasks.list());
+        return !SHUTDOWN;
+    }
+
+    private boolean done(int i) {
+        ui.out(tasks.done(i));
+        return !SHUTDOWN;
+    }
+
+    private boolean todo(String args) throws IncorrectArgumentException {
+        String[] argv = args.split("/");
+        if (argv[0].equals("")) {
+            throw new IncorrectArgumentException("Oops! Missing required arguments: Task Description");
+        } else if (argv.length > 1) {
+            throw new IncorrectArgumentException("Too many arguments!");
+        }
+        ui.out(tasks.addTodo(argv[0]));
+        return !SHUTDOWN;
+    }
+
+    private boolean deadline(String args) throws IncorrectArgumentException {
+        String[] argv = args.split(" /by ");
+        if (argv[0].equals("")) {
+            throw new IncorrectArgumentException(
+                "Oops! Missing required arguments: "
+                .concat(argv.length == 1 
+                    ? "Task Description, by.." 
+                    : "Task Description"));
+        } else if (argv.length < 2) {
+            throw new IncorrectArgumentException("Oops! Missing required arguments: /by..");
+        } else if (argv.length > 2) {
+            throw new IncorrectArgumentException("Too many arguments!");
+        }
+
+        try {
+            String[] textDateTime = argv[1].split(" ");
+            LocalDate date = toDate(textDateTime[0]);
+            LocalTime time = toTime(textDateTime[1]);
+            ui.out(tasks.addDeadline(argv[0], date, time));
+        } catch (DateTimeParseException e) {
+            throw new IncorrectArgumentException("Incorrect Date/Time format!\n     Correct format: yyyy-MM-dd HHmm");
+        }
+
+        return !SHUTDOWN;
+    }
+
+    private boolean event(String args) throws IncorrectArgumentException {
+        String[] argv = args.split(" /at ");
+        if (argv[0].equals("")) {
+            throw new IncorrectArgumentException(
+                "Oops! Missing required arguments: "
+                .concat(argv.length == 1 
+                    ? "Task Description, at.."
+                    : "Task Description"));
+        } else if (argv.length < 2) {
+            throw new IncorrectArgumentException("Oops! Missing required arguments: /at..");
+        } else if (argv.length > 2) {
+            throw new IncorrectArgumentException("Too many arguments!");
+        }
+
+        try {
+            String[] textDateTime = argv[1].split(" ");
+            LocalDate date = toDate(textDateTime[0]);
+            LocalTime time = toTime(textDateTime[1]);
+            ui.out(tasks.addEvent(argv[0], date, time));
+        } catch (DateTimeParseException e) {
+            throw new IncorrectArgumentException("Incorrect Date/Time format!\n     Correct format: yyyy-MM-dd HHmm");
+        }
+        return !SHUTDOWN;
+    }
```

Comment:
Can afford to do a simple and similar javadoc for the 3 different types of tasks